### PR TITLE
Have authorise handle key errors when getting group names from ids

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated Issue.
#### Description of issue:
I got `KeyError` from `cylc.uiserver.authorise.parse_group_ids` because my profile is messed up and keys I wanted were not available.

#### Description of response 
- Added an exception for this case, and created logging for it too.
- Additionally discovered that the logging in `authorise.py` wasn't working, so modified the `app.py` to pass the logger object in.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: cylc/uiserver/authorise.py [hunk 1]
```diff
@@ -387,6 +385,20 @@ def return_site_auth_defaults_for_access_user(
...
+        """Allows get groups to use self.logger if something goes wrong.
+
+        Added to provide a single interface for get_groups to this class, to
+        avoid having to pass the logger to get_groups (and methods it calls).
+        """
+        good_groups, bad_groups = get_groups(user)
+        if bad_groups:
+            self.log.warning(
+                f'{user} has the following invalid groups in their profile '
+                f'{bad_groups}'
```
Comment (oliver-sanders) [inline cylc/uiserver/authorise.py:398]: Worth nothing here that these groups will be ignored for authorisation purposes.

**Review by oliver-sanders — APPROVED** [review]
Looks good, haven't manually tested, MyPy is unhappy for some reason.

File: cylc/uiserver/authorise.py (hunk 1 above)
Comment (wxtim) [reply cylc/uiserver/authorise.py:398]: I was struggling to write these in a way I was happy with.

**Review by datamel — APPROVED** [review]
I have currently working groups so I can not manually check this. I have read the code and am happy with the logic. Thanks @wxtim, and good catch on the issue!

**Comment by oliver-sanders** [comment]
@datamel happy to merge?

**Comment by wxtim** [comment]
Worth saying that it seems to work for Dave.

**Pull request merged**